src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -131,8 +137,9 @@ private function completeDefinition($id, Definition $definition)
      */
     private function populateAvailableTypes()
     {
-        $this->types = array();
+        $this->container->populateAvailableTypes();
```

Comment:
This is here to not have to reload the type map in the container each time we call ``getServiceIdsOfType``.
Otherwise maybe we should have something in the container to know when one of its definition is updated?